nft_ingester/src/api/dapi/response.rs

```diff
@@ -75,6 +79,68 @@ impl From<AssetSignature> for SignatureItem {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default, JsonSchema)]
+#[serde(default)]
+pub struct MasterAssetEditionsInfoModel {
+    pub master_edition_address: String,
+    pub supply: u64,
+    pub max_supply: Option<u64>,
+    pub editions: Vec<AssetEditionInfoModel>,
+
+    #[serde(flatten)]
+    pub pagination: PaginationModel,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default, JsonSchema)]
+pub struct AssetEditionInfoModel {
+    pub mint: String,
+    pub edition_address: String,
+    pub edition: u64,
+}
+impl From<&AssetEditionInfo> for AssetEditionInfoModel {
+    fn from(value: &AssetEditionInfo) -> Self {
+        Self {
+            mint: value.mint.to_string(),
+            edition_address: value.edition_address.to_string(),
+            edition: value.edition,
+        }
+    }
+}
+
+impl From<(MasterAssetEditionsInfo, Pagination)> for MasterAssetEditionsInfoModel {
+    fn from((value, pagination): (MasterAssetEditionsInfo, Pagination)) -> Self {
+        Self {
+            master_edition_address: value.master_edition_address.to_string(),
+            supply: value.supply,
+            max_supply: value.max_supply,
+            editions: value.editions.iter().map(AssetEditionInfoModel::from).collect(),
+            pagination: PaginationModel {
+                total: value.editions.len() as u32,
+                limit: pagination.limit.unwrap(),
+                page: pagination.page,
+                before: pagination.before,
+                after: pagination.after,
+                cursor: pagination.cursor,
+            },
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default, JsonSchema)]
+#[serde(default)]
+pub struct PaginationModel {
+    pub total: u32,
+    pub limit: u32,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub page: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub before: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub after: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub cursor: Option<String>,
+}
+
```

Comment:
i strongly suggest removing the "Model" suffix, and if required, specify fully-qualified paths for rocksdb types. worst case, the "Response" suffix is better than "Model"